src/hooks/useMainProcess.ts

```diff
@@ -257,6 +257,8 @@ export const useMainProcess = () => {
   }, [updateHookStateFromState, offrampingState]);
 
   useEffect(() => {
+    if (wagmiConfig.state.status !== 'connected') return;
+
     (async () => {
       const nextState = await advanceOfframpingState(offrampingState, {
```

Comment:
So for me to understand, the app will "hang" at this stage if whatever step the advance function executes will request a signature, and the user rejects.
Could we add a timeout + refresh if "nothing happens" to avoid this? Potentially [here](https://github.com/pendulum-chain/vortex/blob/2e0dc8d83b6c068ddc34f2df8de449ebd400a06d/src/services/squidrouter/process.ts#L71). Would it make sense? 

Refresh or a modal

Reply:
Sure, it's an option. Can do that.

Reply:
There is just one thing to consider here. There are two situations where our website asks the browser to change the app: 1) when using wallet connect to connect the wallet and 2) when our code asks for signing requests. In both cases the browser shows a popup asking the user to switch the app.

I was testing for a long time yesterday and after some time I always observed something curious: the popup did not appear anymore in either one of the cases. For case 1) I could only make it show up by explicitly pressing the Retry button (see screenshots).

<img src="https://github.com/user-attachments/assets/beee74f1-83b8-4e9b-83f0-6ad31e09c04f" width="300">

<img src="https://github.com/user-attachments/assets/841938a5-8be6-4c3b-b046-0e75b28782aa" width="300">

This looks to me like a security mechanism of Safari in order to protect the user from malicious website behavior. While testing I was repeatedly declining the request to open the app, as this is what I wanted to research. So Safari decided not to show the popup anymore, only in the context of a click handler.

The only way for me to turn on the popup was to change the domain or port, which suggests that Safari blocks by host.

The signing promise will only resolve once the user switched to the app and confirmed there. This can take a while, e.g., looking for a password to log into the app. If during this time we ask repeatedly in the background to switch the app again, it could be that Safari also considers this malicious. Since we start signing requests programmatically and not in click handlers, we risk that the user will never be able to sign anymore.

For that reason I would play it rather safe.

Reply:
Thanks for the thorough testing. So given all this the timer is either a bit pointless or even more confusing for the user. Then let's go with the solution you have right now.
I was thinking more on an absolute timeout in which the app will refresh completely, not just the open request. I don't know if your findings would also apply to this case.

But seems like a rather edge case the scenario in which the person rejects opening a new app but still waits for something to happen.